src/coord/calendar/Calendar.js

```diff
@@ -384,13 +397,20 @@ Calendar.prototype = {
         // ```
         // will get wrong result because of DST. So we should fix it.
         var date = new Date(range[0].time);
+        // 开始的时间是周几
         var startDateNum = date.getDate();
+        // 结束的时间是周几
         var endDateNum = range[1].date.getDate();
+        // 应该结尾的那一天
         date.setDate(startDateNum + allDay - 1);
         // The bias can not over a month, so just compare date.
+        // 不等，则出现DST问题。等则无DST问题
         if (date.getDate() !== endDateNum) {
-            var sign = date.getTime() - range[1].time > 0 ? 1 : -1;
+            // sign记录调快还是调慢了。三月份那次，
+            // I only change '>' to '<'.
+            var sign = date.getTime() - range[1].time < 0 ? 1 : -1;
```

Comment:
Can you explain why you should change this? What does `date.getTime() - range[1].time` mean?